Added Implicit operators for Error and List<Errors>

RHaughton opened the pull request.

altmann commented on src/FluentResults/Results/Result.cs:
```diff
@@ -58,17 +57,16 @@ public Result<TNewValue> Bind<TNewValue>(Func<Result<TNewValue>> bind)
-            if (IsSuccess)
-            {
+
+            if (IsSuccess) {
```
Please change back all this styling changes.

altmann commented on src/FluentResults/Results/Result.cs:
```diff
@@ -176,15 +170,24 @@ public async ValueTask<Result> Bind(Func<ValueTask<Result>> action)
             return result;
         }
+
+        public static implicit operator Result(Error error)
```
Please add some tests for these implicit conversations for Result and Result of T

altmann commented on src/FluentResults/Results/Result.cs:
```diff
@@ -176,15 +170,24 @@ public async ValueTask<Result> Bind(Func<ValueTask<Result>> action)
+            return Fail(error);
+        }
+
+        public static implicit operator Result(List<Error> errors)
```
I hope there is no conflict with the public static implicit operator Result<TValue>(TValue value)
we will see in the tests.

RHaughton commented:
Reverted the style editing (sorry for that).
I just added unit testing for implicit conversion of Error and list of Errors.
Everything seems to work!

eltoncezar commented:
It would be nice if you documented this feature in the README also.

RHaughton commented:
> It would be nice if you documented this feature in the README also.

done

eltoncezar commented:
What you think about converting IReasons instead of just Errors?

RHaughton commented:
Well with errors we know for sure it will be a fail results. Reasons can be errors or successes.

altmann commented:
> What you think about converting IReasons instead of just Errors?

Is it possible to use ```IError``` instead of ```Error```? I think I can remember that implicit conversions and interfaces are not working in .net

Is it possible to use ```IEnumerable<Error or IError>``` instead of ```List<Error or IError>```?

RHaughton commented:
No Unfortunately it doesn't work. I would also prefer the IEnumerable/IError option (I tried it at first).

You can use a List<IError> but the IError alone won't work.

This fails 
`public` static implicit operator Result(IError error) // doesn't work

This works but It is inconsistent with the first Error implicit Operator
`public` static implicit operator Result(List<IError> errors) // works but meh

RHaughton commented:
we could maybe also add an implicit operator for an Error array (Error[]) ?

altmann commented:
> we could maybe also add an implicit operator for an Error array (Error[]) ?

I think a list is okay. I will have a detail look in the next days and merge it.

RHaughton commented:
What do you think of returning and Exception and implicitly convert it to and Fail Result with an exceptionalError?

altmann commented:
Puh, good question. I think its a little bit too much magic. Maybe we can add it in the future if we miss it.
